OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -263,6 +274,9 @@ public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
 				FireDelay = Util.ApplyPercentageModifiers(Weapon.ReloadDelay, modifiers);
 				Burst = Weapon.Burst;
 
+				if (Weapon.ReloadSound != null && Weapon.ReloadSound.Any())
```

Comment:
ScheduleDelayedAction here is wrong, because it will ignore if the actual armement trait ends up disabled between firing and the sound being played.